docarray/array/array/pushpull/__init__.py

```diff
@@ -0,0 +1,244 @@
+import logging
+from abc import abstractmethod
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Iterable, Iterator, List, Optional, Type
+
+from typing_extensions import Protocol
+
+__cache_path__ = Path.home() / '.cache' / 'docarray-v2'
```

Comment:
should rather be `docarray` no ?

Reply:
just an idea, should not we make this `__chache_path__` a function ? WHat happened if I change the my os path during runtime ? should the cache folder change as well ?

Reply:
Yea. I think with a function we can capture better resolution properties. I added the option to change the cache path with env variables into the function.

Hrmmm, changing the path at runtime is a bit weird to me so I disabled it by making it only evaluate once. I think if users want to change cache path at runtime, they should do it in their own code and not count on our DOCARRAY_CACHE env variable.

We can revert [1efc33a](https://github.com/docarray/docarray/pull/1182/commits/1efc33abf5b8e7d669d98807b228ad6567156e53) if we want it to always look at the env variable.